services/placement/placement.go

```diff
@@ -59,13 +59,13 @@ func (p placement) NumInstances() int {
 	return len(p.instances)
 }
 
-func (p placement) Instance(id string) services.PlacementInstance {
+func (p placement) Instance(id string) (services.PlacementInstance, bool) {
 	for _, instance := range p.Instances() {
```

Comment:
@cw9 wouldn't it be most efficient to construct a map of all the instances by ID and then return just:
```go
instance, ok := p.instancesByID[id]
return instance, ok
```

We do a similar thing in the static topology map in M3DB to speed up lookups.